src/main/java/com/pubnub/api/PubNub.java

```diff
@@ -114,8 +114,8 @@ public PubNub(@NotNull PNConfiguration initialConfig) {
         this.mapper = new MapperManager();
         this.telemetryManager = new TelemetryManager();
         this.basePathManager = new BasePathManager(initialConfig);
-        this.retrofitManager = new RetrofitManager(this);
         this.listenerManager = new ListenerManager(this);
+        this.retrofitManager = new RetrofitManager(this);
```

Comment:
I had to change the order of initialisation here. Retrofit c-tor needs ListenerManager not to be null.

Reply:
Maybe pass `listenerManager` explicitly?